### PR TITLE
Fund wallet on all chains script

### PR DESCRIPTION
**Pull request by 0xDEnYO**
# Which Jira task belongs to this PR?

# Why did I implement it this way?

# Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] This pull request is as small as possible and only tackles one problem
- [ ] I have added tests that cover the functionality / test the bug
- [ ] I have updated any required documentation

# Checklist for reviewer (DO NOT DEPLOY and contracts BEFORE CHECKING THIS!!!)

- [ ] I have checked that any arbitrary calls to external contracts are validated and or restricted
- [ ] I have checked that any privileged calls (i.e. storage modifications) are validated and or restricted
- [ ] I have ensured that any new contracts have had AT A MINIMUM 1 preliminary audit conducted on <date> by <company/auditor>

### CODE REVIEW

File: script/tasks/fundNewWalletOnAllChains.ts [hunk 1]
```diff
@@ -128,22 +132,36 @@ const main = defineCommand({
...
+
+    // @DEV: when this script was used the last time this gas estimation was required in order to get the transaction submitted to mainnet
+    //       the data parameter was hardcoded, this should be improved if this code is required permanently
+    // const gas = await publicClient.estimateGas({
+    //   account: fundingWallet,
+    //   to: GAS_ZIP_ROUTER_MAINNET,
+    //   value: amountRequiredNative.toBigInt(),
+    //   data: '0x6e553f6500000000ff393e0f36608c9415140a1f103b0d1e491c1d371134fe29f6f99233000000000000000000000000d38743b48d26743c0ec6898d699394fbc94657ee',
+    // })
+    // console.log(`Gas estimated: ${gas}`)
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/tasks/fundNewWalletOnAllChains.ts:164]: no it is commented out for a reason and should stay like this

File: script/tasks/fundNewWalletOnAllChains.ts [hunk 2]
```diff
@@ -153,12 +171,20 @@ const main = defineCommand({
...
-    const txHash = await walletClient.writeContract(result.request)
+    const txHash = await walletClient.writeContract({
+      ...result.request,
+    })
     console.log(`Transaction successfully submitted: ${txHash}`)
+
+    const transaction = await publicClient.waitForTransactionReceipt({
+      hash: txHash,
+    })
+    console.dir(transaction, { depth: null, colors: true })
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/tasks/fundNewWalletOnAllChains.ts:187]: not required

File: script/utils/viemScriptHelpers.ts [hunk 3]
```diff
@@ -1,27 +1,31 @@
-import { Chain, defineChain, getAddress } from 'viem'
+import { Chain, defineChain, getAddress, PublicClient } from 'viem'
```
Comment (ezynda3) [inline script/utils/viemScriptHelpers.ts:1]: Is `PublicClient` actually being used anywhere?

File: script/deploy/safe/confirm-safe-tx.ts
Comment (ezynda3) [inline script/deploy/safe/confirm-safe-tx.ts:1]: We might want to check the other safe scripts as well to make sure this doesn't break them

**Review by ezynda3 — CHANGES_REQUESTED** [review]

File: script/deploy/safe/confirm-safe-tx.ts
Comment (0xDEnYO) [reply script/deploy/safe/confirm-safe-tx.ts:1]: I couldnt find any other script that uses this type when I updated it. Are you aware of any other usage?

Comment (ezynda3) [reply script/deploy/safe/confirm-safe-tx.ts:1]: Nope. Should be gtg.

**Review by ezynda3 — APPROVED** [review]

File: script/utils/viemScriptHelpers.ts [hunk 4]
```diff
@@ -1,27 +1,30 @@
...
+  explorerUrl: string
+  explorerApiUrl: string
+  multicallAddress: string
+  safeApiUrl: string
+  safeAddress: string
+  gasZipChainId: number
+  id: string
+}
+
+const networks: NetworksObject = networksConfig
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/utils/viemScriptHelpers.ts:27]: no need to

**Pull request merged**